src/pytest_memray/marks.py

```diff
@@ -46,15 +51,84 @@ def long_repr(self) -> str:
         return f"Test was limited to {max_memory_str} but allocated {total_memory_str}"
 
 
-def limit_memory(
-    limit: str, *, _allocations: list[AllocationRecord]
-) -> Optional[_MemoryInfo]:
+@dataclass
+class _LeakedInfo:
```

Comment:
There might be some refactor potential here, considering that `_MemoryInfo` and `_LeakedInfo` share the exact same set of fields, and the implementations of `section` and `long_repr` are only very slightly different for the two.